packages/support/src/helpers.php

```diff
@@ -163,15 +163,11 @@ function generate_search_column_expression(string $column, ?bool $isSearchForced
     {
         $driverName = $databaseConnection->getDriverName();
 
-        if (Str::lower($column) !== $column) {
-            $column = match ($driverName) {
-                'pgsql' => (string) str($column)->wrap('"'),
-                default => $column,
-            };
-        }
-
         $column = match ($driverName) {
-            'pgsql' => "{$column}::text",
+            'pgsql' => (string) str($column)
+                ->explode('.')
```

Comment:
I think this needs to consider JSON fields, e.g. `->`.

For example, we wouldn't want to end up with ...

```
"data ->> 'name'"::text
```

If you update your branch you'll see my recent addition to fix JSON field searching, so hopefully this can be incorporated into that.

My update also provided some tests that you can borrow from to cover this update.

Please see https://github.com/filamentphp/filament/blob/3.x/packages/support/src/helpers.php#L173-L183

And https://github.com/filamentphp/filament/blob/3.x/tests/src/Support/helpersTest.php#L47-L79

Reply:
@glennjacobs that was merged in now, and JSON fields aren't being changed. Maybe they should be though?